students/JeroenSweerts/Lesson5/mailroom3.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+donations = {}
+donations['papa'] = [100, 5, 15]
+donations['mama'] = [12, 200, 2, 66]
+donations['bompa'] = [1000]
+donations['bobonne'] = [500, 500]
+donations['onbekende'] = [1000000]
+
+
+def thankyou():
+    """If the user types ‘list’, show them a list of the donor names and re-prompt
+    If the user types a name not in the list, add that name to the data structure
+    and use it.
+    If the user types a name in the list, use it.
+    Once a name has been selected, prompt for a donation amount.
+    Turn the amount into a number – it is OK at this point for the program to
+    crash if someone types a bogus amount.
+    Once an amount has been given, add that amount to the donation history of
+    the selected user.
+    Finally, use string formatting to compose an email thanking the donor for their
+    generous donation. Print the email to the terminal and return to the original
+    prompt."""
+    while True:
+        name = input('Please enter a name or type "list" to see a list of names > ')
+        if name == 'list':
+            print(donations.keys())
+        else:
+            break
+    if name not in donations.keys():
+        donations[name] = []
+    try:
+        donation = input('Please enter a donation amount > ')
+        donations[name].append(int(donation))
+    except ValueError:
+        while not donation.isdigit():
+            print("You didn't enter a valid amount.")
+            donation = input('Please enter a donation amount > ')
+        donations[name].append(int(donation))
+    d = {'name':name, 'donation':donation}
+    print("Dear {name} , thank you very much for your generous donation of {donation}. Looking forward receiving even more money next time.".format(**d))
+    print(donations.keys())
+    print(donations.values())
+
+
+def createfile(name):
```

Comment:
all of your function names should use underscore to separate words: `create_file`, `thank_you`, `thank_you_everyone`